### PR TITLE
Date range management FAD-5334

### PR DESCRIPTION
**Pull request by jonambas**
- Selecting "Custom" opens Datepicker
- Datepicker & Date fields stay in sync
  - Date Fields are validated on blur
  - Dates must be valid dates
  - From must be before To
  - To must be before now
  - Must match `MM/DD/YY` and `h:mma`
- Reverts to previous dates if invalid

kapture 2017-08-22 at 16 09 16

### CODE REVIEW

File: src/actions/reportFilters.js [hunk 1]
```diff
@@ -1,7 +1,12 @@
+import { getRelativeDates } from 'helpers/metrics';
+
 export function setRelativeTime(payload) {
```
Comment (jonambas) [inline src/actions/reportFilters.js]: Think we can technically use one action for exact and relative time, but figured I'd separate them out for when we get to auto-refreshing

**Comment by jasonrhodes** [comment]
First initial thought is: we can't hardcode the `MM/DD/YY` format ... internationally that format is bizarre and confusing.

It would be nice if we could use a format like `15-Aug 2017`, but if not that, then i'd vote for `YYYY-MM-DD` ... i wish there was an easy way to do this with locale formatting in the browser but i was looking a few weeks ago and it seemed _almost_ there but not quite

**Comment by jasonrhodes** [comment]
We need to switch up the x-axis for "last hour" -- less ticks and showing every quarter hour somehow. 

Current app x-axis for last hour:
screen shot 2017-08-23 at 2 49 50 pm

**Comment by jasonrhodes** [comment]
Can we add a "cancel" button on the date picker pop up that resets everything to the way it was before it was opened? so reset datepicker internal state, reset relative drop down if it was previously selected, close datepicker popup (pressing "esc" should do all these things too)

**Comment by jasonrhodes** [comment]
Pressing "enter" while in the inputs should do the same as "Apply" -- a nice to have would be having enter pressed at any time while that picker is open will apply the settings ... maybe with the exception of while you're in between date selections because during that time "enter" locks in the "to" date ...

File: src/pages/reports/components/DateFilter.js [hunk 2]
```diff
@@ -45,6 +27,15 @@ class DateFilter extends Component {
     window.removeEventListener('keydown', this.handleEsc);
   }
 
+  componentWillReceiveProps(nextProps) {
+    this.syncStateToProps(nextProps);
+  }
+
+  // Sets local state from reportFilters redux state - need to separate to handle pre-apply state
+  syncStateToProps = ({ filter }) => {
```
Comment (jonambas) [inline src/pages/reports/components/DateFilter.js]: should be `syncPropstoState`

**Pull request merged**